source/server/guarddog_impl.cc

```diff
@@ -42,7 +51,30 @@ GuardDogImpl::GuardDogImpl(Stats::Scope& stats_scope, const Server::Configuratio
           Stats::StatNameManagedStorage("server.watchdog_mega_miss", stats_scope.symbolTable())
               .statName())),
       dispatcher_(api.allocateDispatcher("guarddog_thread")),
-      loop_timer_(dispatcher_->createTimer([this]() { step(); })), run_thread_(true) {
+      loop_timer_(dispatcher_->createTimer([this]() { step(); })),
+      events_to_actions_([&](const Server::Configuration::Main& config) -> EventToActionsMap {
+        EventToActionsMap map;
+
+        // We should be able to share the dispatcher since guard dog's lifetime
+        // should eclipse those of actions.
+        Configuration::GuardDogActionFactoryContext context = {api, *dispatcher_};
+
+        const auto& actions = config.wdActions();
+        for (const auto& action : actions) {
+          if (action.event() == WatchDogAction::UNKNOWN) {
+            throw ProtoValidationException("WatchDogAction specified with UNKNOWN event", action);
```

Comment:
Please a PGV annotation on the proto for enum defined only. That will make this impossible and checked by the schema.